shell/packages/sandstorm-db/db.js

```diff
@@ -1651,27 +1661,29 @@ _.extend(SandstormDb.prototype, {
     });
   },
 
-  setPreinstallAppAsDownloading: function (appId) {
-    this.collections.settings.update({ _id: "preinstalledApps", "value.appId": appId },
+  setPreinstallAppAsDownloading: function (appId, packageId) {
+    this.collections.settings.update(
+      { _id: "preinstalledApps", "value.appId": appId, "value.packageId": packageId },
       { $set: { "value.$.status": "downloading" } });
   },
 
-  setPreinstallAppAsReady: function (appId) {
-    this.collections.settings.update({ _id: "preinstalledApps", "value.appId": appId },
-      { $set: { "value.$.status": "ready" } });
+  setPreinstallAppAsReady: function (appId, packageId) {
+    // This function both sets the appId as ready and updates the packageId for the given appId
+    // Setting the packageId is especially useful in installer.js, as it always ensures the
+    // latest installed package will be set as ready.
+    this.collections.settings.update(
+      { _id: "preinstalledApps", "value.appId": appId },
+      { $set: { "value.$.status": "ready", "value.$.packageId": packageId } });
   },
 
-  ensureAppPreinstall: function (appId) {
+  ensureAppPreinstall: function (packageId, appId) {
```

Comment:
Weird that this function takes the args in `packageId, appId` order then passes them to another function that expects them in `appId, packageId` order (`setPreinstallAppAsReady()`).